qiskit/quantum_info/operators/symplectic/sparse_pauli_op.py

```diff
@@ -128,11 +128,9 @@ def __repr__(self):
 
     def __eq__(self, other):
         """Check if two SparsePauliOp operators are equal"""
-        return (
-            super().__eq__(other)
-            and np.allclose(self.coeffs, other.coeffs)
-            and self.paulis == other.paulis
-        )
+        if not super().__eq__(other):
+            return False
+        return np.allclose((self - other).simplify().coeffs, [0])
```

Comment:
It would be good to check the performance characteristics of this.  The dict-based `simplify` has better asymptotic scaling than it used to, so this becomes more feasible, but this makes a relatively simple operation involve probably 3 `SparsePauliOp` allocations (in `-other`, `self._add(-other)` and `simplify`).